imap/jmap_mail.c

```diff
@@ -1077,9 +1077,14 @@ static char *_emailbodies_to_html(struct emailbodies *bodies, const struct buf *
     return buf_releasenull(&buf);
 }
 
-static const char *_guid_from_id(const char *msgid)
+static const char *_guid_from_id(struct conversations_state *cstate,
+                                 const char *emailid)
 {
-    return msgid + 1;
+    static char guidrep[2*MESSAGE_GUID_SIZE+1];
+
+    int r = conversations_lookup_jmapid(cstate, emailid + 1, guidrep);
```

Comment:
non-actionable comment: I get why this is done here, but it sucks to have a db call introduced into a function that formerly only had the cost of simple pointer arithmetic.

Reply:
Already the emailid+1 was really a code smell and I figured doing that in a single function at least contains that. It's not a big deal to call into conversations now, I guess, at least I don't see this being called in a hot loop (or is it?)

Reply:
It shouldn't be